solr/core/src/java/org/apache/solr/util/BinaryUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+
+public class BinaryUtils {
```

Comment:
There is similar code in 

```
solr/solrj/src/java/org/apache/solr/client/solrj/impl/BinaryRequestWriter.java:87:  public static class BAOS extends ByteArrayOutputStream {
```

Might make sense to use this class there as well?

Reply:
I made the change to combine them in c486782d67f9c927809968105f21cc2ff78ab3af, and it ended up being a bit more intrusive then I expected.  All tests pass.  Can you give it a quick look?